concourse/enumerator.py

```diff
@@ -267,15 +267,42 @@ def __init__(self, repository_url:str, cfg_set):
         self.cfg_set = not_none(cfg_set)
         concourse_cfg = cfg_set.concourse()
         job_mapping_set = cfg_set.job_mapping(concourse_cfg.job_mapping_cfg_name())
-
-        # hack: use first mapping that matches
         org = self._repository_url.path.lstrip('/').split('/')[0]
-        for job_mapping in job_mapping_set.job_mappings().values():
-            if org in [o.org_name() for o in job_mapping.github_organisations()]:
-                self.job_mapping = job_mapping
-                break
-        else:
+        matching_mappings = list(self._find_matching_job_mappings(
```

Comment:
as discussed yesterday, it makes not too much sense to use a generator, just to put it into a list at the only user

consider the following pattern:
```
def find_element(condition):
  candidate = None
  for element in elements:
    if condition(element):
      if candiate:
        # signal error - not unambigous
      candidate = element
  return candiate # optionally check if found and raise, depending on your coding style preference
```